src/test/java/org/sonarsource/sonarlint/ls/mediumtests/CFamilyMediumTests.java

```diff
@@ -48,6 +52,21 @@ static void initialize() throws Exception {
       "productVersion", "0.1"));
   }
 
+  @Override
+  protected void setUpFolderSettings(Map<String, Map<String, Object>> folderSettings) {
+    super.setUpFolderSettings(folderSettings);
+    client.readyForTestsLatch = new CountDownLatch(1);
+  }
+
+  @Override
+  protected void verifyConfigurationChangeOnClient() {
```

Comment:
I wonder if it would make sense to extract this implementation and share it between other tests (4 in total) that override this `verifyConfigurationChangeOnClient` method

Reply:
Actually, I'm wondering whether this should be the default behavior for all medium tests.

At the moment, I believe that it is OK to keep it duplicated, and maybe have a mob session in the squad.